telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt

```diff
@@ -40,21 +44,32 @@ object TelemetryGenerator {
                     .addParameter("name", String::class)
                     .build()
             )
+            .addFunction(FunSpec.builder("toString").addModifiers(KModifier.OVERRIDE).returns(String::class).addStatement("return name").build())
+            .addKdoc(item.description)
+
         item.allowedValues!!.forEach { enumValue ->
             enum.addEnumConstant(
                 enumValue.toString().toUpperCase().filterInvalidCharacters(), TypeSpec.anonymousClassBuilder()
                     .addSuperclassConstructorParameter("%S", enumValue.toString())
                     .build()
             )
         }
-        enum.addFunction(FunSpec.builder("toString").addModifiers(KModifier.OVERRIDE).returns(String::class).addStatement("return name").build())
-        enum.addFunction(
-            FunSpec.builder("from")
-                .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
-                .addParameter("type", Any::class)
-                .addStatement("return values().firstOrNull { it.name == type.toString() }").build()
-        )
-        enum.addKdoc(item.description)
+
+        val companion = TypeSpec.companionObjectBuilder()
+            .addFunction(
+                FunSpec.builder("from")
+                    .returns(ClassName("", item.name.toTypeFormat()).copy(nullable = true))
```

Comment:
To me `from` is effectively a constructor so it should never return null. Either you get back an object or it blows up.

Reply:
That was my initial thought, but I'm hesitant to throw as a result of telemetry. I can easily imagine that we make a change and forget to change the telemetry which would actually affect a workflow

Reply:
Had a conversation offline, reverted this change and made it able to throw again

Reply:
Not sure what the convo was, and I agree that generally we don't want Telemetry to break the general flow - but this should only be the case if there's a code-problem at build time - right? so it should be OK in this case.